### PR TITLE
test: include tests for user task filter services and increase coverage variable tests

### LINKED ISSUE #21845: [Root Task] Search Task by Variable
```[tasklist]
### Tasks
- [ ] https://github.com/camunda/camunda/issues/21843
- [ ] https://github.com/camunda/camunda/issues/21846
- [ ] https://github.com/camunda/camunda/issues/21848
- [ ] https://github.com/camunda/camunda/issues/21849
```

### PR DESCRIPTION
**Pull request by nathansandi**
## Description

Unit test implementation for the follow Pull-Request:
https://github.com/camunda/camunda/pull/19273

closes #https://github.com/camunda/camunda/issues/19210

### CODE REVIEW

**Review by romansmirnov — CHANGES_REQUESTED** [review]
@nathansandi, the test coverage has to be increased here. IMO, we should do the following

1. Test all filter options
2. Test variations of specific filter options (like passing only one value or multiple values, ...)
3. Test the sorting options

File: service/src/main/java/io/camunda/service/transformers/filter/UserTaskFilterTransformer.java [hunk 1]
```diff
@@ -87,19 +84,6 @@ public List<String> toIndices(final UserTaskFilter filter) {
...
-      final var transformer = getVariableValueFilterTransformer();
-      final var queries =
-          variableFilters.stream()
-              .map(transformer::apply)
-              .map((q) -> hasChildQuery("variable", q))
-              .collect(Collectors.toList());
-      return and(queries);
-    }
-    return null;
-  }
```
Comment (romansmirnov) [inline service/src/main/java/io/camunda/service/transformers/filter/UserTaskFilterTransformer.java:101]: Why this change? 🤔

Comment (nathansandi) [reply service/src/main/java/io/camunda/service/transformers/filter/UserTaskFilterTransformer.java:101]: I decided make isolated the variables search for User tasks, the reasons:
- 2 different indices, would bring performance problems once I would apply joins and nested queries on ES
- I am going filter task by variables similar in the way is made today, first performa the Task Search based on attributes, and later perform Variable search by the Task results, I am going perform this over the business logic later during the implementation  of the controller, and it will be handled in a separate Pull-Request

Comment (nathansandi) [reply service/src/main/java/io/camunda/service/transformers/filter/UserTaskFilterTransformer.java:101]: Note I have the issue separated over here: https://github.com/camunda/tasklist/issues/4967  
So independent of chosen solution, It will be handled in a separated Pull Request

Comment (romansmirnov) [reply service/src/main/java/io/camunda/service/transformers/filter/UserTaskFilterTransformer.java:101]: @nathansandi, thanks for the explanations.

Some feedback from my side:
* This PR focuses on adding test cases, and deleting that code caused confusion on my side because the PR does not provide much context on why that method was deleted.
* I recommend extracting such additional changes in a separate PR to ensure a good and focused review.

Some additional comments to the shared comment:

> If you prefer, I can move this to the service layer and add the variables filter there.

The business logic must be implemented in the service, not the REST layer. The REST layer must remain thin and clear. This ensures that we can easily test the REST layer and don't need to deal with testing business logic there. Of course, the business logic must be tested in the service layer—there is no way around it. But I hope that the current infrastructure makes writing test cases easier. If not, let's see how we can extend the test infrastructure.

> 1- The current method for getting variables is not functional for the User Task.
> 2- Using two different indices could cause performance issues when applying joins and nested queries in Elasticsearch.
>
> I plan to filter tasks by variables in a similar way to how it's done today. First, I'll perform the Task Search based on attributes, and then I'll perform the Variable search based on the Task results. I'll handle this in the business logic during the implementation of the controller.

To be honest, I disagree with the statement that applying joins and/or nested queries causes performance issues. In other words, did you execute any benchmarks or some research that underlines that conclusion?

When reviewing the current implementation, I assume that it is less efficient (for example, in terms of memory consumption) and probably also less performant than applying joins and/or nested queries in ES/OS (because the implementation's number of requests - to ES/OS - grows with the number of total hits at least linearly).

I strongly recommend revisiting that approach and implementing a proper solution to support the use case.

**Review by romansmirnov — APPROVED** [review]
Since this PR mainly focuses on adding test cases, I am hereby approving it. The other topics are not related to this PR in that sense.

File: service/src/main/java/io/camunda/service/transformers/filter/UserTaskFilterTransformer.java (hunk 1 above)
Comment (nathansandi) [reply service/src/main/java/io/camunda/service/transformers/filter/UserTaskFilterTransformer.java:101]: Hey @romansmirnov related to the has_child/join queries performance issues: https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-has-child-query.html 

You can find a lot of forum posts from ES community also reporting strong performance issues related to nested querie

"We don’t recommend using multiple levels of relations to replicate a relational model. Each level of relation adds an overhead at query time in terms of memory and computation. For better search performance, denormalize your data instead." 

In compare to multiple call executions, I see the follow advantages:
- Single Call with Joins: Elasticsearch is not designed for relational data and complex joins. Performing joins or using has_child queries can be resource-intensive and slow, especially on large datasets. It requires Elasticsearch to perform internal operations that can significantly impact performance.
- Multiple Calls: By making multiple calls, you can break down the problem into smaller, more manageable parts. This can help distribute the load and reduce the complexity of each individual query, leading to better performance and scalability.
- Custom Logic: Performing joins in application code gives you more control over the join logic, allowing for more complex and tailored operations that might be cumbersome to express in Elasticsearch DSL.
- Complex join operation can consume a lot of memory and CPU resources on the Elasticsearch cluster, potentially impacting other operations and overall cluster health. 

I have performed a benchmark test on the past implementation on Tasklist, because we had the similar implementation over there. Unfortunately it was 8 months ago, I could not find the result here, but I can do an extra PR with `has_child` implementation in order to compare and remake this test. 

We can continue discuss on : https://github.com/camunda/camunda/pull/19691
